python_modules/libraries/dagster-fivetran/dagster_fivetran/translator.py

```diff
@@ -67,7 +67,44 @@ def to_fivetran_connector_table_props_data(self) -> Sequence[FivetranConnectorTa
         """Method that converts a `FivetranWorkspaceData` object
         to a collection of `FivetranConnectorTableProps` objects.
         """
-        raise NotImplementedError()
+        data: List[FivetranConnectorTableProps] = []
+
+        for connector_id, connector_data in self.connectors_by_id.items():
```

Comment:
In airlift, from the raw API data we retrieved, we constructed a "lookup" object which built up the actual usable asset data from a set of cacheable properties. See https://github.com/dagster-io/dagster/blob/0a0c77b42d8138e8703db3b94fa8381b86e16036/examples/experimental/dagster-airlift/dagster_airlift/core/serialization/compute.py#L114 for an example of what I mean.

Since this is over the "cacheable" boundary, I'm wondering if there's a likelihood if it being called a bunch, and if so I think the cacheable structure likely makes sense. At the very least, I think that the conversion to `FivetranConnectorTableProps` should probably be cached, but it might also make sense to cache the stuff from 73-83 here as properties if there's potential for reuse (but if not, then maybe that's not worth the effort)

Reply:
That's a good point. This method is meant to be used only in `FivetranWorkspaceDefsLoader.defs_from_state`, see implement in next PR https://github.com/dagster-io/dagster/pull/25807.

But I think making this a cached property `all_fivetran_connector_table_props` makes sense.

Reply:
I cached the method in [b272d07](https://github.com/dagster-io/dagster/pull/25797/commits/b272d0728101ec8167c190608551077fea4a81f1)